### DISCUSSION TITLE
Updating from older version

### DISCUSSION THREAD

**Discussion by johanboekhoven** [aminalaee/sqladmin#526] [Q&A]
I'm updating from 0.8.0 to the latest release, and there have been some understandable breaking changes. 
But I have a somewhat hard time porting this over to working template code:

```py
{{ model_view.get_attr_value(model, model_view.get_model_atr("asset_id")) }}
```

I know get_attr_value() is replaced with get_prop_value(), but how do I get specific data from the model on a detail page. I want to use specific database fields to add extra data to the details page.

Thanks!

**Comment by johanboekhoven** [comment]
Worked around it using a for loop, but it doesn't feel great:

```py
{% for detail in model_view._details_props %}

    {% if detail[0] == 'asset_id' %} 

        {% set asset_id, asset_id_fmt = model_view.get_detail_value(model, detail[1]) %}
        
        .. template code here
```

I would to just set a variable somewhere by name and pick it up where ever I need it.

**Reply by aminalaee** [reply]
I'm not 100% sure I got what you are looking for, but maybe you need this?
https://github.com/aminalaee/sqladmin/blob/a1844d267f0a4627e4d2fca3b5ec59491d160f7b/sqladmin/models.py#L923
The templates are using this and the benefit is that it will work with your custom formatters.

**Reply by aminalaee** [reply]
And about breaking changes, you are right. I'm trying to be more careful to make methods more private.

**Reply by johanboekhoven** [reply]
Thanks! As you can see in my second post, I'm using the get_detail_value() function, but it expects a key/val pair of data, and it would be great if I can just feed it an property name, because now I need to loop through a for loop and filter it out, making the template a bit of a mess. 
Can I construct the MODEL_PROPERTY in some deterministic way perhaps?

**Reply by aminalaee** [reply]
As I think you have seen `MODEL_PROPERTY` is not special, it's either ColumnProperty or RelationshipProperty.
Not sure if you need both, but I think `_column_props` would give you the list (not tested and not recommended) for this.

I think the main issue is that the approach is not correct, I think ideally you should be able to define custom properties and hybrid_properties to be able to control this, overriding the template like this will not be very efficient.

**Reply by johanboekhoven** [reply]
Thanks @aminalaee, I understand it's rather unconventional, I just have a very customized/specialized details page for a specific type of model. Whit very specific form buttons on it, that call other API's etc. The form data needs to be filled with specific data. Hence I just want to filter out very specific fields as variables to be used in the template. Hence the need to directly access certain properties from the model. But I'll figure it out, thanks for your input!
